starknet-foundry/crates/cast/src/starknet_commands/account/deploy.rs

```diff
@@ -1,5 +1,114 @@
+use crate::helpers::constants::OZ_CLASS_HASH;
+use anyhow::{anyhow, bail, Result};
+use camino::Utf8PathBuf;
+use cast::{get_network, handle_rpc_error, parse_number, print_formatted, wait_for_tx};
 use clap::Args;
+use starknet::accounts::AccountFactoryError;
+use starknet::accounts::{AccountFactory, OpenZeppelinAccountFactory};
+use starknet::core::types::FieldElement;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+use starknet::signers::{LocalWallet, SigningKey};
 
 #[derive(Args, Debug)]
 #[command(about = "Deploy an account to the Starknet")]
-pub struct Deploy {}
+pub struct Deploy {
+    /// Name of the account to be deployed
+    #[clap(short, long)]
+    pub name: String,
+
+    /// Max fee for the transaction
+    #[clap(short, long)]
+    pub max_fee: FieldElement,
+}
+
+pub async fn deploy(
+    provider: &JsonRpcClient<HttpTransport>,
+    path: Utf8PathBuf,
+    name: String,
+    network: &str,
+    max_fee: FieldElement,
+) -> Result<FieldElement> {
+    let network = get_network(network)?.get_value();
+
+    let contents = std::fs::read_to_string(path.clone())?;
+    let mut items: serde_json::Value =
+        serde_json::from_str(&contents).expect("failed to parse json file");
+
+    if items[network].is_null() {
+        bail!("Provided network does not have any accounts defined")
+    }
+    if items[network][&name].is_null() {
+        bail!("Account with provided name does not exist")
+    }
+
+    let private_key = SigningKey::from_secret_scalar(
+        parse_number(
+            items
+                .get(network)
+                .and_then(|network| network.get(&name))
+                .and_then(|name| name.get("private_key"))
+                .and_then(serde_json::Value::as_str)
+                .ok_or_else(|| anyhow!("Couldn't get private key from accounts file"))?,
+        )
+        .expect("Couldn't parse private key"),
+    );
+
+    let factory = OpenZeppelinAccountFactory::new(
+        parse_number(OZ_CLASS_HASH).expect("Couldn't parse OpenZeppelins account class hash"),
+        provider.chain_id().await.expect("Couldn't get chain id"),
+        LocalWallet::from_signing_key(private_key),
+        provider,
+    )
+    .await?;
+
+    let deployment = factory.deploy(
+        parse_number(
+            items
+                .get(network)
+                .and_then(|network| network.get(&name))
+                .and_then(|name| name.get("salt"))
+                .and_then(serde_json::Value::as_str)
+                .ok_or_else(|| anyhow!("Couldn't get salt from accounts file"))?,
```

Comment:
Is salt required? I think (but am not sure) there might be a scenario where there is no salt

Reply:
Ok, I have checked and it seems this is indeed required field :+1: